Lib/test/test_builtin.py

```diff
@@ -1511,6 +1513,126 @@ def test_construct_singletons(self):
             self.assertRaises(TypeError, tp, 1, 2)
             self.assertRaises(TypeError, tp, a=1, b=2)
 
+
+class TestBreakpoint(unittest.TestCase):
+    def test_breakpoint(self):
+        with patch('pdb.set_trace') as mock:
+            breakpoint()
+        mock.assert_called_once()
+
+    def test_breakpoint_with_breakpointhook_set(self):
+        call_status = 'Not called'
+        def my_breakpointhook():
+            nonlocal call_status
+            call_status = 'Called'
```

Comment:
You might use a mock to get arguments: assert_called_once_with(...).

Reply:
I'm not quite sure what you're suggesting here.  There are other tests to ensure that the right positional and keyword arguments are passed through.  This test just ensures that a custom `sys.breakpointhook` gets called at all.  (I like focused unit tests.)

Reply:
I think haypo is suggesting code like:

```python
myhook = mock.MagicMock()

... code under test...

myhook.assert_called_once_with()
```

instead of manual tracking of the call status.

Reply:
@merwok ah, thanks!  I like that.